Products/zms/zpt/ZMSMetamodelProvider/manage_main.zpt

```diff
@@ -791,8 +792,92 @@ $(function(){
 }
 // Add Execute Button for TAL-Code-Regeneration om a ZPT-Method named standard_html
 var zmi_code_default_btn = '<i onclick="set_zmi_code_default(event)" title=\"Replace by Default-Code-Template\" class=\"zmi-code-default icon-repeat fas fa-redo-alt text-primary text-center\" style=\"cursor:pointer;display:block;position:absolute;text-align:right !important;right:0.75em;padding:0.5em;\"></i>';
+
+// Execute on DOM-Ready
 $(function(){
 	$('#meta_properties input[name="attr_type_standard_html"] ~ .zmi-code').prepend( zmi_code_default_btn);
+
+	// ++++++++++++
+	// Add rows to tables #meta_properties or #meta_languages on button click
+	// ++++++++++++
+	let new_row_counter = 0;
+	// Get number of language terms
+	let lang_count = Object.keys(JSON.parse(ZMI.prototype.getCachedValue('get_lang_dict'))).length
+
+	// Add click event function to add-buttons
+	$(".row_insert .btn-add").click(function(){
+		new_row_counter++;
+
+		// Where to insert the new row 
+		let $where_insert = $(this).closest('tr');
+
+		// Set variables
+		let table_id = $(this).closest('table').attr('id'); // meta_properties or meta_languages
+		let new_row_name = `new_row_${table_id}_${new_row_counter}`;
+		let new_btn_html = `
+			<input type="hidden" name="old_ids:list" value="new${new_row_counter}">
+			<span class="btn btn-default btn-sm mr-1" style="color:#999" 
+				onclick="javascript:$(this).closest('tr').remove()">
+				<i class="fas fa-times"></i>
+			</span>
+			`;
+
+		// Clone(true) to get a deep copy including select options
+		let $new_row = $where_insert.clone(true);
+
+		// Process table cells of the clone like "old" row
+		$new_row.find('td').each(function() {
+			$(this).find('input,select,textarea').each(function() {
+				let tagname = $(this).prop('tagName');
+				let newname =  $(this).attr('name').split('_')[1];
+
+				// #### NAMES: Set new form elemnt names according to table_id
+				switch (table_id) {
+					case 'meta_properties':
+						switch (newname.includes(':')) {
+							case true: // Checkboxes (:int-suffix)
+								$(this).attr('name', `attr_${newname.split(':')[0]}_new${new_row_counter}:${newname.split(':')[1]}`);
+								break
+							default: // String Inputs
+								$(this).attr('name', `attr_${newname}_new${new_row_counter}`);
+						};
+					case 'meta_languages':
+						$(this).attr('name',$(this).attr('name').replace('_0', '_' + (lang_count + new_row_counter)));

```

Comment:
don't we need an individual row-counter for each table?

Reply:
I do not think so, because counter is need for the ex-ante (temporarily) name unification (that is "corrected" by the ZMSMetaobjManager.manage_changeProperties() applying the old_id-mechanism). 
BTW: Maybe we can get rid of the counter at all, when generally changing all the new row's form names on-blur according to the input-value of the meta-id field. The current JS code may be considered as too long ;-)